### ISSUE TITLE
Improvements: OS3.9 Boing Bag 2, and WinUAE fast emulated machine MHz detection 

### ISSUE THREAD

**Issue by Hexaae** [shred/identify#1]
Reminder for possible improvements of identify.library, when used in Scout or other Amiga tools:
- OS3.9 BBag 2 correct detection
- new hardware (Vampire etc.?)
- better MHz detection under faaast WinUAE emulation (currently showing it as 060/0Mhz or sometimes 7Mhz...)
....
😉

**Comment by shred** [comment]
Progress! :smile: 

- All "classic" AmigaOS versions up to AmigaOS 3.2 are now detected.
- Some new hardware is detected (especially from Individual Computer, MNT, as well as UAE extensions).
- Emulated FPU clock detection has been improved. However for very fast emulations the clock frequency can still be 0MHz, as the busy loop finishes below timer resolution. The measured frequency on emulated machines is useless anyway, as the JIT will optimize the busy loop, resulting in way to high frequencies.

The changes will be part of the next version, which will be released after I finished some more testing.

About Vampire: Do you own such a nice machine? It might be possible to detect the SAGA chipset (and the 68080 CPU), but I would need someone who can test it (or who would donate me a Vampire, hehehe... :laughing: ).

**Comment by Hexaae** [comment]
No Vampire sorry... Write on [EAB](http://eab.abime.net/index.php), for sure someone has got it.

**Comment by polluks** [comment]
Just use the official code
````
#ifndef AFB_68080
#define AFB_68080 10
#define AFF_68080 (1 << AFB_68080)
#endif
````

**Comment by shred** [comment]
Thank you, @polluks. That's really easy. The next version will get 68080 and SAGA chipset detection.

**Comment by shred** [comment]
[Release 38.0](https://github.com/shred/identify/releases/tag/v38.0) can be found in the release section now, and should also be available on AmiNet soon.

The requested problems in this issue should be resolved with that release. If not, please feel free to reopen this bug, or create a new one.

Thank you for your patience! It was a very long time... :wink:

**Comment by Hexaae** [comment]
... just wondering about minor issue n.3 (better MHz detection on faaast Amigas) if it could report a "?" instead of nonsense number (0-7Mhz...), if won't be too much work to code 😁

**Comment by shred** [comment]
Starting with V38, only a real Amiga hardware with real Motorola CPU gives an accurate clock frequency now. On emulators or FPGA based systems, "0 MHz" is always returned as frequency, meaning "clock is not available". `ListExp` is then hiding the CPU/FPU clock. I have no influence on how other software is rendering the "0 MHz" value.

Do you have an example system or tool where a nonsense number is given?

**Comment by shred** [comment]
Thinking about that again, I actually can give "N/A" instead of "0 MHz" on _some_ tools. I will change that.

**Comment by Hexaae** [comment]
Great! :)
For example Scout http://m68k.aminet.net/package/util/moni/Scout_os3 opening System window....

**Comment by shred** [comment]
Depending on how Scout fetches the clock frequency, it should be fixed with the next Identify release now (V39.1 or V40.0, it's not decided yet). If Scout should then still give "0 MHz", you probably need to open a bug report at the Scout developers, and hope for a fix.

**Comment by Hexaae** [comment]
![image](https://user-images.githubusercontent.com/8591462/153852628-838baa4c-5b1b-4c87-88ce-066ee5122a14.png)
Nice the N/A, I like it! ;)
